starcheck/calc_ccd_temps.py

```diff
@@ -109,23 +111,34 @@ def get_ccd_temps(oflsdir, outdir='out',
     :param oflsdir: products directory
     :param outdir: output directory for plots
     :param json_obsids: file-like object or string containing JSON of
-                        starcheck Obsid objects
-    :param model_spec: xija ACA model specification
-    :param run_start_time: Chandra.Time date used as a reference time to determine initial
-                     seed state with temperature telemetry.  The initial seed state will
-                     be at the end of available telemetry that is also before run_start_time
-                     and before the beginning of backstop cmds.
+                        starcheck Obsid objects (default='<oflsdir>/starcheck/obsids.json')
+    :param model_spec: xija ACA model spec file (default=package aca_spec.json)
+    :param char_file: starcheck characteristics file (default=package characteristics.yaml)
+    :param run_start_time: Chandra.Time date, clock time when starcheck was run,
+                     or a user-provided value (usually for regression testing).
     :param verbose: Verbosity (0=quiet, 1=normal, 2=debug)
+    :param kwargs: extra args, including test_rltt and test_sched_stop for testing
+
     :returns: JSON dictionary of labeled dwell intervals with max temperatures
     """
     if not os.path.exists(outdir):
         os.mkdir(outdir)
 
+    module_dir = Path(__file__).parent
+    if model_spec is None:
+        model_spec = str(module_dir / 'data' / 'aca_spec.json')
+    if char_file is None:
+        char_file = str(module_dir / 'data' / 'characteristics.yaml')
+
+    if json_obsids is None:
+        # Only happens in testing, so use existing obsids file in OFLS dir
+        json_obsids = Path(oflsdir, 'starcheck', 'obsids.json').read_text()
```

Comment:
I suppose if one uses a local oflsdir one would not need to presume that the tested changes in local starcheck weren't causing diffs in the obsids.json.

Reply:
I don't understand this.

Reply:
If we were to change something about the timing of the obsids (saved in that obsids.json file) with a change in starcheck, the new 'test code' here in calc_ccd_temps.py probably wouldn't get the right answer to test a new starhceck release candidate if you use /data/mpcrit1/mplogs oflsdirs for input.  I think I was saying, that your test fixes in calc_ccd_temps.py are great but they don't solve all integration testing issues with the Perl pieces of starcheck.pl.